react-charts/donut-chart/step4.md

```diff
@@ -1,31 +1,57 @@
-Now we'll add tooltips to the donut chart...
+Add a legend to the donut chart.
 
-## Task
+1) <strong>Make sure the `App.js` file is still open.</strong>
 
-1) Make sure the App.js file is still open
+2) <strong>Locate the code for the `<Chart>` component.</strong>
 
-2) Locate the code that looks like the following:
+It should look like this:
 
 <pre class="file">
 &lt;ChartDonut
+  constrainToVisibleArea
   data={[{ x: &#39;Cats&#39;, y: 35 }, { x: &#39;Dogs&#39;, y: 55 }, { x: &#39;Birds&#39;, y: 10 }]}
+  labels={({ datum }) =&gt; `${datum.name}: ${datum.y}%`}
   subTitle=&quot;Pets&quot;
   title=&quot;100&quot;
   height={230}
   width={350}
 /&gt;
 </pre>
 
-3) Add the following properties to that section:
+3) <strong>Add a `padding` property to the component.</strong>
+
+This `padding` property adds more padding to accommodate the legend.
+
+Copy this code to the editor:
 
 <pre class="file" data-target="clipboard">
-constrainToVisibleArea
-labels={({ datum }) =&gt; `${datum.x}: ${datum.y}%`}
+padding={{
+  bottom: 20,
+  left: 20,
+  right: 140,
+  top: 20
+}}
 </pre>
 
-- The `constrainToVisibleArea` property will alter the position of the tooltip so that it exactly fits within the rendered SVG
 
-4) Once the preview reloads - it should look like this:
-<img src="donut-chart/assets/tooltips.png" alt="Chart with tooltips" style="box-shadow: rgba(3, 3, 3, 0.2) 0px 1.25px 2.5px 0px;" />
+4) <strong>Add the `LegendOrientation` and `legendPosition` properties to the component.</strong>

```

Comment:
```suggestion
4) <strong>Add the `legendOrientation` and `legendPosition` properties to the component.</strong>
```

Reply:
Split this up to be just adding the `legendOrientation` property.
Then have a new step for the `legendPosition` property.
And then a following step for the `legendData` property.

See area-chart, step 6 for more information